griptape/drivers/embedding/base_embedding_driver.py

```diff
@@ -12,23 +12,25 @@
 class BaseEmbeddingDriver(ExponentialBackoffMixin, ABC):
     """
     Attributes:
+        model: The name of the model to use.
         tokenizer: An instance of `BaseTokenizer` to use when calculating tokens.
-        chunker: An instance of `BaseChunker` to use when chunking long strings.
     """
 
-    tokenizer: BaseTokenizer = field(kw_only=True)
+    model: str = field(kw_only=True)
+    tokenizer: BaseTokenizer | None = field(kw_only=True)
     chunker: BaseChunker = field(init=False)
 
     def __attrs_post_init__(self) -> None:
-        self.chunker = TextChunker(tokenizer=self.tokenizer)
+        if self.tokenizer:
+            self.chunker = TextChunker(tokenizer=self.tokenizer)
 
     def embed_text_artifact(self, artifact: TextArtifact) -> list[float]:
         return self.embed_string(artifact.to_text())
 
     def embed_string(self, string: str) -> list[float]:
         for attempt in self.retrying():
             with attempt:
-                if self.tokenizer.count_tokens(string) > self.tokenizer.max_tokens:
+                if self.tokenizer and self.tokenizer.count_tokens(string) > self.tokenizer.max_tokens:
                     return self._embed_long_string(string)
                 else:
                     return self.try_embed_chunk(string)
```

Comment:
Is this meaningful behavior if the tokenizer is not specified? What use case does it enable?

Reply:
A Tokenizer is only needed when embedding long strings that need to be chunked up. Some embedding models do not provide tokenization information (max tokens, tokenization algorithm, etc) so there is no great way to build a tokenizer. By making it optional, we hand the reigns to the developer to embed strings of a length they determine.